pkg/controllers/node/termination/terminator/suite_test.go

```diff
@@ -127,53 +129,33 @@ var _ = Describe("Eviction/Queue", func() {
 				Labels:         testLabels,
 				MaxUnavailable: &intstr.IntOrString{IntVal: 1},
 			})
-			ExpectApplied(ctx, env.Client, pod)
-			Expect(queue.Evict(ctx, terminator.NewQueueKey(pod, node.Spec.ProviderID))).To(BeTrue())
+			ExpectApplied(ctx, env.Client, pod, node)
+			Expect(queue.Evict(ctx, pod)).To(BeNil())
 			Expect(recorder.Calls(events.Evicted)).To(Equal(1))
 		})
 		It("should return a NodeDrainError event when a PDB is blocking", func() {
-			ExpectApplied(ctx, env.Client, pdb, pod)
-			Expect(queue.Evict(ctx, terminator.NewQueueKey(pod, node.Spec.ProviderID))).To(BeFalse())
+			ExpectApplied(ctx, env.Client, pdb, pod, node)
+			Expect(queue.Evict(ctx, pod)).ToNot(BeNil())
 			Expect(recorder.Calls(events.FailedDraining)).To(Equal(1))
 		})
 		It("should fail when two PDBs refer to the same pod", func() {
 			pdb2 := test.PodDisruptionBudget(test.PDBOptions{
 				Labels:         testLabels,
 				MaxUnavailable: &intstr.IntOrString{IntVal: 0},
 			})
-			ExpectApplied(ctx, env.Client, pdb, pdb2, pod)
-			Expect(queue.Evict(ctx, terminator.NewQueueKey(pod, node.Spec.ProviderID))).To(BeFalse())
+			ExpectApplied(ctx, env.Client, pdb, pdb2, pod, node)
+			Expect(queue.Evict(ctx, pod)).ToNot(BeNil())
 			ExpectMetricCounterValue(terminator.NodesEvictionRequestsTotal, 1, map[string]string{terminator.CodeLabel: "500"})
 		})
 		It("should ensure that calling Evict() is valid while making Add() calls", func() {
```

Comment:
Why did we delete this test?

Reply:
How does that work? We're still not calling Reconcile in this test so I'm not clear how this exercises these things in parallel

Reply:
Yeah, I think we really need to get to a place where pods successfully evict and then we are adding pods to the queue through the Evict call at the same time -- we basically just need to test that the queue is thread-safe -- adding and deleting at the same time doesn't cause any issues